src/utils/get-embedded-app-url.ts

```diff
@@ -1,12 +1,11 @@
-import http from 'http';
-
 import * as ShopifyErrors from '../error';
 import {ConfigInterface} from '../base-types';
+import {NormalizedRequest} from '../runtime/http';
 
 import {createSanitizeHost} from './shop-validator';
 
 export function createGetEmbeddedAppUrl(config: ConfigInterface) {
-  return (request: http.IncomingMessage): string => {
+  return (request: NormalizedRequest): string => {
```

Comment:
As per Slack chat, this should be hidden away from the dev ... so, using `AdapterArgs` as the param type, the dev need only pass in (`{rawRequest: req}`) for whatever platform they're using and we can hide the conversion internally.